Update DNS providers

KizunaH opened the pull request:
**What's new in this PR?**

- Added new providers. (small providers also included)
_Did you considered anything when add small providers?
Yes. has project website, alexa rank and past uptime._
- Added AdGuard DNS non-filtering mode.
- Fixed CleanBrowsing IPs. stamps and added new protocol.
- Changed providers order for users to recognize local providers easily.
(Again, it's not possible to bring all sprats near to sharks.)
(Do not categorize by country or put small providers to separate location)
- Add Quad9 with ECS support  #438 
- Update Quad101 DNS and BlahDNS #436
- [PR] Add OpenDNS DoH #531 (only has DOH for family mode. Discard)
- [PR] Fix CIRA Canadian Shield #532 (doesn't follow template. Discard)
- Adjusted DNS providers names.
- Fixed some DNS mode names.
- Fixed other IPs, stamps and tables.

KizunaH commented:
https://github.com/AdguardTeam/AdguardKnowledgeBase/pull/552

admitrevskiy reviewed (changes requested):
@KizunaH Hi!
Thank you for such a great contribution to the growth of this section of the knowledge base.

Several fixes are needed.
Moreover, please replace `Location` column with a header if all the values in this column are the same

e.g.

| Protocol | Address   | Location      |            |
|---------|-----------|-------------|-------|
| a            |     b            |        c            |  ...       |
| d            |     e            |        c            |  ...       |
| f            |     g             |        c            |  ...       |

Should be replaced with

####c

| Protocol | Address   |                     |
|---------|-----------|-------------|
| a            |     b            |            ...       |
| d            |     e            |            ...       |
| f            |     g            |             ...       |

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -144,12 +186,24 @@ Blocks phishing, spam and malicious domains
 | DNS, IPv4      | `185.228.168.9` and `185.228.169.9`                | <a href="sdns://AAAAAAAAAAAADTE4NS4yMjguMTY4Ljk">Add to AdGuard</a> |
 | DNS, IPv6      | `2a0d:2a00:1::2` and `2a0d:2a00:2::2`               | <a href="sdns://AAAAAAAAAAAAEFsyYTBkOjJhMDA6MTo6Ml0">Add to AdGuard</a> |
 | DNS-over-HTTPS | `https://doh.cleanbrowsing.org/doh/security-filter/` | <a href="sdns://AgMAAAAAAAAAAAAVZG9oLmNsZWFuYnJvd3Npbmcub3JnFS9kb2gvc2VjdXJpdHktZmlsdGVyLw">Add to AdGuard</a> |
+| DNS-over-TLS | `tls://security-filter-dns.cleanbrowsing.org` | <a href="sdns://AwAAAAAAAAAAAAAVdGxzOi8vZG5zLmFkZ3VhcmQuY29t">Add to AdGuard</a> |
```
[1]

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -275,38 +331,55 @@ OpenDNS servers that provide adult content blocking
+| DNS, IPv6      | `2620:fe::fe` IP: `2620:fe::fe:9`| <a href="sdns://AAAAAAAAAAAADVsyNjIwOmZlOjpmZV0">Add to AdGuard</a> |
+| DNSCrypt, IPv4 | Provider: `2.dnscrypt-cert.quad9.net` IP: `9.9.9.9:8443`| <a href="sdns://AQMAAAAAAAAADDkuOS45Ljk6ODQ0MyBnyEe4yHWM0SAkVUO-dWdG3zTfHYTAC4xHA2jfgh2GPhkyLmRuc2NyeXB0LWNlcnQucXVhZDkubmV0">Add to AdGuard</a> |
+| DNSCrypt, IPv6 |  Provider: `2.dnscrypt-cert.quad9.net` IP: `[2620:fe::fe]:8443`  | <a href="sdns://AQMAAAAAAAAAElsyNjIwOmZlOjpmZV06ODQ0MyBnyEe4yHWM0SAkVUO-dWdG3zTfHYTAC4xHA2jfgh2GPhkyLmRuc2NyeXB0LWNlcnQucXVhZDkubmV0">Add to AdGuard</a> |
+| DNS-over-HTTPS | `dns.quad9.net/dns-query`  | <a href="sdns://AgEAAAAAAAAAAAANZG5zLnF1YWQ5Lm5ldAovZG5zLXF1ZXJ5">Add to AdGuard</a> |
```
[2]

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -275,38 +331,55 @@ OpenDNS servers that provide adult content blocking
+| DNS, IPv6      | `2620:fe::9` IP: `2620:fe::fe:9`  | <a href="sdns://AAAAAAAAAAAADFsyNjIwOmZlOjo5XQ">Add to AdGuard</a> |
+| DNSCrypt, IPv4 | Provider: `2.dnscrypt-cert.quad9.net` IP: `149.112.112.9:8443`  | <a href="sdns://AQEAAAAAAAAAEjE0OS4xMTIuMTEyLjk6ODQ0MyBnyEe4yHWM0SAkVUO-dWdG3zTfHYTAC4xHA2jfgh2GPhkyLmRuc2NyeXB0LWNlcnQucXVhZDkubmV0">Add to AdGuard</a> |
+| DNSCrypt, IPv6 |  Provider: `2.dnscrypt-cert.quad9.net` IP: `[2620:fe::9]:8443`  | <a href="sdns://AQMAAAAAAAAAEVsyNjIwOmZlOjo5XTo4NDQzIGfIR7jIdYzRICRVQ751Z0bfNN8dhMALjEcDaN-CHYY-GTIuZG5zY3J5cHQtY2VydC5xdWFkOS5uZXQ">Add to AdGuard</a> |
+| DNS-over-HTTPS | `dns9.quad9.net/dns-query`  | <a href="sdns://AgEAAAAAAAAAAAANZG5zLnF1YWQ5Lm5ldAovZG5zLXF1ZXJ5">Add to AdGuard</a> |
```
[3]

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -275,38 +331,55 @@ OpenDNS servers that provide adult content blocking
+| DNS-over-HTTPS | `dns.quad9.net/dns-query`  | <a href="sdns://AgEAAAAAAAAAAAANZG5zLnF1YWQ5Lm5ldAovZG5zLXF1ZXJ5">Add to AdGuard</a> |
+| DNS-over-TLS   | `tls://9.9.9.9`            | <a href="sdns://AwAAAAAAAAAAAAANdGxzOi8vOS45LjkuOQ">Add to AdGuard</a> |
+
+#### Standard 2
```
[4]

admitrevskiy replied:
It's weird :)
Could you replace sdns stamp for plain IPv4 server (it's for 9.9.9.9) with a new one for `149.112.112.9`
Just to make it at least different with the previous one

KizunaH replied:
Updated.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -275,38 +331,55 @@ OpenDNS servers that provide adult content blocking
+| DNS, IPv6      | `2620:fe::10` IP: `2620:fe::fe:10`  | <a href="sdns://AAAAAAAAAAAADVsyNjIwOmZlOjoxMF0">Add to AdGuard</a> |
+| DNSCrypt, IPv4 | Provider: `2.dnscrypt-cert.quad9.net` IP: `9.9.9.10:8443`  | <a href="sdns://AQMAAAAAAAAADTkuOS45LjEwOjg0NDMgZ8hHuMh1jNEgJFVDvnVnRt803x2EwAuMRwNo34Idhj4ZMi5kbnNjcnlwdC1jZXJ0LnF1YWQ5Lm5ldA">Add to AdGuard</a> |
+| DNSCrypt, IPv6 |  Provider: `2.dnscrypt-cert.quad9.net` IP: `[2620:fe::fe:10]:8443`  | <a href="sdns://AQMAAAAAAAAAFVsyNjIwOmZlOjpmZToxMF06ODQ0MyBnyEe4yHWM0SAkVUO-dWdG3zTfHYTAC4xHA2jfgh2GPhkyLmRuc2NyeXB0LWNlcnQucXVhZDkubmV0">Add to AdGuard</a> |
+| DNS-over-HTTPS | `dns10.quad9.net/dns-query`  | <a href="sdns://AgEAAAAAAAAAAAAPZG5zMTAucXVhZDkubmV0Ci9kbnMtcXVlcnk">Add to AdGuard</a> |
```
[5]

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -417,142 +839,323 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
-| DNS, IPv4 | Provider: `dns.switch.ch` IP: `130.59.31.248` | <a href="sdns://AAAAAAAAAAAADTEzMC41OS4zMS4yNDg">Add to AdGuard</a> |
-| DNS, IPv6 | Provider: `dns.switch.ch` IPv6: `2001:620:0:ff::2`| <a href="sdns://AAAAAAAAAAAAElsyMDAxOjYyMDowOmZmOjoyXQ">Add to AdGuard</a>  |
-| DNS-over-TLS | Hostname: `dns.switch.ch` IP: `130.59.31.248`  and IPv6: `2001:620:0:ff::2` | <a href="sdns://AwAAAAAAAAAAAAATdGxzOi8vZG5zLnN3aXRjaC5jaA">Add to AdGuard</a>  |
+| DNS-over-TLS, IPv4 | Hostname: `dot.ffmuc.net` IP: `195.30.94.28` | <a href="sdns://AwAAAAAAAAAAAAATdGxzOi8vZG90LmZmbXVjLm5ldA">Add to AdGuard</a> |
```
[6]

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -417,142 +839,323 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+
+| Protocol       | Address                                            |                |
+|----------------|----------------------------------------------------|----------------|
+| DNS, IPv4      | `185.95.218.42` and `185.95.218.43`          | <a href="sdns://AAAAAAAAAAAADTE4NS45NS4yMTguNDI">Add to AdGuard</a> |
```
[7]

KizunaH replied:
Removed plain rows. Provider said they do not offer unencrypted service and those IPs relevant for advanced configuration environment such as restrictive firewall setup.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -417,142 +839,323 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
 
 | Protocol       | Address                                            |                |
 |----------------|----------------------------------------------------|----------------|
+| DNS, IPv4      | `116.202.176.26`             | <a href="sdns://AAAAAAAAAAAADjExNi4yMDIuMTc2LjI2">Add to AdGuard</a> |
```
[8]

KizunaH replied:
Updated.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -417,142 +839,323 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+| DNS-over-HTTPS, IPv4 | Hostname: `https://ibksturm.synology.me/dns-query` IP: `178.82.102.190` | <a href="sdns://AgcAAAAAAAAAACA-GhoPbFPz6XpJLVcIS1uYBwWe4FerFQWHb9g_2j24OBRpYmtzdHVybS5zeW5vbG9neS5tZQovZG5zLXF1ZXJ5">Add to AdGuard</a> |
+| DNSCrypt, IPv4 | Provider: `2.dnscrypt-cert.ibksturm` IP: `83.77.85.7:8443` | <a href="sdns://AQcAAAAAAAAADzgzLjc3Ljg1Ljc6ODQ0MyDBz1dQALBbwmxiH17PmqJWCs6_AH6-yzp_9LIN4LQ57hgyLmRuc2NyeXB0LWNlcnQuaWJrc3R1cm0">Add to AdGuard</a> |
+| DNSCrypt, IPv6 | Provider: `2.dnscrypt-cert.ibksturm` IP: `[2a02:1205:5055:de60:b26e:bfff:fe1d:e19b]:8443` | <a href="sdns://AQcAAAAAAAAALlsyYTAyOjEyMDU6NTA1NTpkZTYwOmIyNmU6YmZmZjpmZTFkOmUxOWJdOjg0NDMgwc9XUACwW8JsYh9ez5qiVgrOvwB-vss6f_SyDeC0Oe4YMi5kbnNjcnlwdC1jZXJ0Lmlia3N0dXJt">Add to AdGuard</a> |
+| Anonymized DNS IPv4 | IP: `85.5.93.230:8443` | <a href="sdns://gRA4NS41LjkzLjIzMDo4NDQz">Add to AdGuard</a> |
```
[9]

admitrevskiy replied:
Could you combine PR with this type of upstreams with PR for DoQ upstreams? We will add this to KB as soon as clients can support it

KizunaH replied:
Anonymized DNS rows removed and not intend to move to separate PR. that's a hobby project and only the provider belong for that in this list then adguard may able to consider to add when necessary.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -417,142 +839,323 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+PI-DNS currently provide a DoH endpoint delivered through Cloudflare’s CDN. This can be used if you don’t want to specify any specific server to connect to. Make sure you understand Cloudflare’s privacy policy before using this endpoint.
 
-<a name="evilvibes.com"></a>
+Endpoint: `https://doh.pi-dns.com/dns-query`
```
[10]

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -275,38 +331,55 @@ OpenDNS servers that provide adult content blocking
+| DNS, IPv6      | `2620:fe::11` IP: `2620:fe::fe:11`  | <a href="sdns://AAAAAAAAAAAADVsyNjIwOmZlOjoxMV0">Add to AdGuard</a> |
+| DNSCrypt, IPv4 | Provider: `2.dnscrypt-cert.quad9.net` IP: `9.9.9.11:8443`  | <a href="sdns://AQMAAAAAAAAADTkuOS45LjExOjg0NDMgZ8hHuMh1jNEgJFVDvnVnRt803x2EwAuMRwNo34Idhj4ZMi5kbnNjcnlwdC1jZXJ0LnF1YWQ5Lm5ldA">Add to AdGuard</a> |
+| DNSCrypt, IPv6 |  Provider: `2.dnscrypt-cert.quad9.net` IP: `[2620:fe::11]:8443`  | <a href="sdns://AQMAAAAAAAAAElsyNjIwOmZlOjoxMV06ODQ0MyBnyEe4yHWM0SAkVUO-dWdG3zTfHYTAC4xHA2jfgh2GPhkyLmRuc2NyeXB0LWNlcnQucXVhZDkubmV0">Add to AdGuard</a> |
+| DNS-over-HTTPS | `dns11.quad9.net/dns-query`   | <a href="sdns://AgEAAAAAAAAAAAAPZG5zMTEucXVhZDkubmV0Ci9kbnMtcXVlcnk">Add to AdGuard</a> |
```
[11]

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -319,6 +392,29 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+| Protocol       | Address                                            |                |
+|----------------|----------------------------------------------------|----------------|
+| DNS, IPv4      | `130.59.31.248` and `130.59.31.251`                | <a href="sdns://AAAAAAAAAAAADTEzMC41OS4zMS4yNDg">Add to AdGuard</a>  |
+| DNS, IPv6      | `2001:620:0:ff::2` and `2001:620:0:ff::3` | <a href="sdns://AAAAAAAAAAAAETIwMDE6NjIwOjA6ZmY6OjJd">Add to AdGuard</a> |
```
[7]

admitrevskiy replied:
This stamp is for `2001:620:0:ff::2]`. It should be `[2001:620:0:ff::2]

KizunaH replied:
Updated. (already has on kb site as it)

KizunaH replied:
Updated. (already has on kb site as it. Mistake copy pasted the same)

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -319,6 +392,29 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+
+| Protocol       | Address                                            |                |
+|----------------|----------------------------------------------------|----------------|
+| DNS, IPv4      | `216.146.35.35` and `216.146.35.36`                | <a href="sdns://AAAAAAAAAAAADTIxNi4xNDYuMzUuMzU">Add to AdGuard</a> |
```
[8]

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -330,26 +426,96 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
 |----------------|----------------------------------------------------|----------------|
-| DNS, IPv4      | `8.26.56.26` and `8.20.247.20`                     | <a href="sdns://AAAAAAAAAAAACjguMjYuNTYuMjY">Add to AdGuard</a> |
-| DNSCrypt, IPv4 | Provider: `2.dnscrypt-cert.shield-2.dnsbycomodo.com` IP: `8.20.247.2`   | <a href="sdns://AQAAAAAAAAAACjguMjAuMjQ3LjIg0sJUqpYcHsoXmZb1X7yAHwg2xyN5q1J-zaiGG-Dgs7AoMi5kbnNjcnlwdC1jZXJ0LnNoaWVsZC0yLmRuc2J5Y29tb2RvLmNvbQ">Add to AdGuard</a> |
+| DNS, IPv4      | `208.91.112.220` and `80.85.69.54`                 | <a href="sdns://AAAAAAAAAAAADjIwOC45MS4xMTIuMjIw">Add to AdGuard</a> |
```
[9]

KizunaH replied:
No response from company. Provider removed.

KizunaH replied:
No response from company. Provider removed.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -330,26 +426,96 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+
+[Comss.one DNS](https://www.comss.ru/) is a fast and secure DNS servers with protection against ads, tracking and phishing.
+
+| Protocol       | Address                             | Location      |                |
```
[10]

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -417,142 +839,323 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+
+| Protocol       | Address                                            |                |
+|----------------|----------------------------------------------------|----------------|
+| DNS, IPv4      | `158.64.1.29`             | <a href="sdns://AAAAAAAAAAAACzE1OC42NC4xLjI5">Add to AdGuard</a> |
```
Updated. It seems not plain.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -417,142 +839,323 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
 | Protocol       | Address                                            |                |
 |----------------|----------------------------------------------------|----------------|
-| DNSCrypt, IPv4 | Provider: `2.dnscrypt-cert.captnemo.in` IP: `139.59.48.222:4434`    | <a href="sdns://AQQAAAAAAAAAEjEzOS41OS40OC4yMjI6NDQzNCAFOt_yxaMpFtga2IpneSwwK6rV0oAyleham9IvhoceEBsyLmRuc2NyeXB0LWNlcnQuY2FwdG5lbW8uaW4">Add to AdGuard</a> |
+| DNS, IPv4      | `176.9.199.158`              | <a href="sdns://AAAAAAAAAAAADTE3Ni45LjE5OS4xNTg">Add to AdGuard</a> |
```
It's working. check again.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -417,142 +839,323 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
 | Protocol       | Address                                            |                |
 |----------------|----------------------------------------------------|----------------|
-| DNS, IPv4      | `104.155.237.225` and `104.197.28.121`             | <a href="dns://AAAAAAAAAAAADjA0LjE1NS4yMzcuMjI1">Add to AdGuard</a> |
+| DNS, IPv4      | `104.155.237.225` and `104.197.28.121`             | <a href="AAAAAAAAAAAADzEwNC4xNTUuMjM3LjIyNQ">Add to AdGuard</a> |
```
Updated. it has on kb site as it.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -362,6 +528,180 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+|----------------|----------------------------------------------------|----------------|
+| DNS, IPv4      | `119.29.29.29` and `119.28.28.28`    | <a href="sdns://AAAAAAAAAAAADDExOS4yOS4yOS4yOQ">Add to AdGuard</a> |
+| DNS-over-HTTPS | `https://doh.pub/dns-query` | <a href="sdns://AgcAAAAAAAAAAAAHZG9oLnB1YgovZG5zLXF1ZXJ5">Add to AdGuard</a> |
+| DNS-over-HTTPS | `https://dns.pub/dns-query` | <a href="sdns://AgcAAAAAAAAAAAAHZG5zLnB1YgovZG5zLXF1ZXJ5">Add to AdGuard</a> |
```
Plain seems only available for DNS provider's country. I'll check further.

KizunaH replied:
It's working (chinese provider). may be blocked in your country :(

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -362,6 +528,180 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+
+<a name="nawala-childprotection-dns"></a>
+
+### Nawala Childprotection DNS
```
DNS servers may be temporally down (that has on kb site already)

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -362,6 +528,180 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+| DNS, IPv4      | `94.130.180.225` and `78.47.64.161`             | <a href="sdns://AAAAAAAAAAAADjk0LjEzMC4xODAuMjI1">Add to AdGuard</a> |
+| DNS, IPv6      | `2a01:4f8:1c0c:40db::1` and `2a01:4f8:1c17:4df8::1`            | <a href="sdns://AAAAAAAAAAAAF1syYTAxOjRmODoxYzBjOjQwZGI6OjFd">Add to AdGuard</a> |
+| DNSCrypt, IPv4 | Provider: `dnsforfamily.com` IP: `94.130.180.225`| <a href="sdns://AQIAAAAAAAAADjk0LjEzMC4xODAuMjI1ILtn1Ada3rLi6VNcj4pB-I5eHBqFzFbs_XFRHG-6KenTEGRuc2ZvcmZhbWlseS5jb20">Add to AdGuard</a> |
+| DNSCrypt, IPv6 |  Provider: `dnsforfamily.com` IP: `[2a01:4f8:1c0c:40db::1]`| <a href="sdns://sdns://AQIAAAAAAAAAF1syYTAxOjRmODoxYzBjOjQwZGI6OjFdIKeNqJacdMufL_kvUDGFm5-J2r4yS94vn4S5ie-o8MCMEGRuc2ZvcmZhbWlseS5jb20">Add to AdGuard</a> |
```
Updated.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -362,6 +528,180 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+
+| Protocol       | Address                                            |                |
+|----------------|----------------------------------------------------|----------------|
+| DNS, IPv4      | `95.216.24.230`            | <a href="sdns://AAAAAAAAAAAADTk1LjIxNi4yNC4yMzA">Add to AdGuard</a> |
```
Updated. It seems not plain.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -362,6 +528,180 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+| DNSCrypt, IPv6 | Provider: `2.dnscrypt-cert.blahdns.com` IP: `2001:19f0:7001:3259:5400:02ff:fe71:0bc9:8443`| `Japan` | <a href="sdns://AQMAAAAAAAAALlsyMDAxOjE5ZjA6NzAwMTozMjU5OjU0MDA6MDJmZjpmZTcxOjBiYzldOjg0NDMgU4ToFEMUKT5W3RsUCh7xcq1HvboXmciVcpSVPQNOtccbMi5kbnNjcnlwdC1jZXJ0LmJsYWhkbnMuY29t">Add to AdGuard</a> |
+
+| Protocol       | Address                             | Location      |                |
+|----------------|-------------------------------------|---------------|----------------|
```
1. The problem these IP addresses unable to use as plain but it is helpful for advanced network environments and DNS project site also provide separate IPv4 and IPv6 stamps.
2. Updated

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -362,6 +528,180 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+
+| Protocol       | Address                             | Location      |                |
+|----------------|-------------------------------------|---------------|----------------|
+| DNS-over-TLS, IPv4 | Hostname: `dot-jp.blahdns.com`  IP: `45.32.55.94` | `Japan` | <a href="sdns://AwAAAAAAAAAAAAAYdGxzOi8vZG90LWpwLmJsYWhkbnMuY29t">Add to AdGuard</a> |
```
Updated.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -362,6 +528,180 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+| Protocol       | Address                             | Location      |                |
+|----------------|-------------------------------------|---------------|----------------|
+| DNS-over-TLS, IPv4 | Hostname: `dot-fi.blahdns.com`  IP: `95.216.212.177` | `Finland` | <a href="sdns://AwAAAAAAAAAAAAAYdGxzOi8vZG90LWRlLmJsYWhkbnMuY29t">Add to AdGuard</a> |
+| DNS-over-HTTPS, IPv4 | Hostname: `https://doh-fi.blahdns.com/dns-query` IP: `95.216.212.177` | `Finland` | <a href="sdns://AgMAAAAAAAAADjE1OS42OS4xOTguMTAxABJkb2gtZGUuYmxhaGRucy5jb20KL2Rucy1xdWVyeQ">Add to AdGuard</a> |
```
Updated.

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -362,6 +528,180 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+
+| Protocol       | Address                             | Location      |                |
+|----------------|-------------------------------------|---------------|----------------|
+| DNS-over-TLS, IPv4 | Hostname: `dot-fi.blahdns.com`  IP: `95.216.212.177` | `Finland` | <a href="sdns://AwAAAAAAAAAAAAAYdGxzOi8vZG90LWRlLmJsYWhkbnMuY29t">Add to AdGuard</a> |
```
Updated.

KizunaH commented:
@admitrevskiy sorry for there some incorrect details and past contributors works also appear as my own due re-arrange tables, extend details and other modifications. I fixed all you mentioned issues. let me know if something missing.

Added one more commit because i didn't noticed hidden 16 conversations(github).

>Moreover, please replace Location column with a header if all the values in this column are the same

Updated. I followed existing template eg: [dns-provider](https://kb.adguard.com/en/general/dns-providers#evilvibes-com). that's the reason.

**What's new?**
Added DNS over Quic for AdGuard DNS
(check if the stamp is correct)

admitrevskiy commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -362,6 +535,180 @@ Regular and DNS-over-TLS servers which provide protection from phishing and spyw
+| Protocol       | Address                             |                |
+|----------------|-------------------------------------|----------------|
+| DNS-over-TLS, IPv4 | Hostname: `tls://dot-fi.blahdns.com`  IP: `95.216.212.177` | <a href="sdns://AwAAAAAAAAAAAAAYdGxzOi8vZG90LWZpLmJsYWhkbnMuY29t">Add to AdGuard</a> |
+| DNS-over-HTTPS, IPv4 | Hostname: `https://doh-fi.blahdns.com/dns-query` IP: `95.216.212.177` | <a href="sdns://AgMAAAAAAAAAAAASZG9oLWRlLmJsYWhkbnMuY29tCi9kbnMtcXVlcnk">Add to AdGuard</a> |
```
This stamp is for `doh-de.blahdns.com` :)
DNSCrypt stamps are also incorrect

KizunaH replied:
Updated.

admitrevskiy reviewed (changes requested):
@KizunaH there are still couple of mistakes in the stamps
Moreover, please move not supported protocols to a separate PR: we will add support ASAP

KizunaH commented on 01.general/17.dns-providers/docs.en.md:
```diff
@@ -70,6 +98,21 @@ These servers provide the Default features + Blocking adult websites + Safe sear
 | DNSCrypt, IPv6 | Provider: `2.dnscrypt.family.ns2.adguard.com` IP: `[2a00:5a60::bad2:0ff]:5443`| <a href="sdns://AQIAAAAAAAAAGlsyYTAwOjVhNjA6OmJhZDI6MGZmXTo1NDQzIIwhF6nrwVfW-2QFbwrbwRxdg2c0c8RuJY2bL1fU7jUfITIuZG5zY3J5cHQuZmFtaWx5Lm5zMi5hZGd1YXJkLmNvbQ">Add to AdGuard</a> |
 | DNS-over-HTTPS | `https://dns-family.adguard.com/dns-query` | <a href="sdns://AgcAAAAAAAAAAAAWZG5zLWZhbWlseS5hZGd1YXJkLmNvbQovZG5zLXF1ZXJ5">Add to AdGuard</a> |
 | DNS-over-TLS | `tls://dns-family.adguard.com` | <a href="sdns://AwAAAAAAAAAAAAAcdGxzOi8vZG5zLWZhbWlseS5hZGd1YXJkLmNvbQ">Add to AdGuard</a> |
+| DNS-over-QUIC | `quic://dns-family.adguard.com` | <a href="sdns://AwAAAAAAAAAAAAAdcXVpYzovL2Rucy1mYW1pbHkuYWRndWFyZC5jb20">Add to AdGuard</a> |
```
See https://github.com/AdguardTeam/AdGuardHome/issues/2049
However i removed DoQ rows and not intend for move to separate PR because that's simple thing, someone in adguard may able to do that. (only adguard dns offer that currently)

admitrevskiy reviewed (approved):
@KizunaH Hi!
Everything's fine with the stamps

We need a review from @Dariamag now

KizunaH commented:
Ping @Dariamag Can you review this soon and merge?
New PRs coming and all conflict with this due update using old providers list.
